api/v1beta1/types.go

```diff
@@ -84,11 +84,32 @@ type FilterByNeutronTags struct {
 	NotTagsAny []NeutronTag `json:"notTagsAny,omitempty"`
 }
 
+// SecurityGroupFilter describes the data needed to identify an SG to use. If ID is provided it is required that all other fields are unset.
```

Comment:
I wonder if we could make this validation similar to [the image filter](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1968/files#diff-964de9482de3e9bdd98a2e17c86bbc25650afa8b719a4c42a2a0014fc3b1869dR32) one for simplicity purposes. I don't have a strong opinion

Reply:
I needed to split it to avoid going over the CEL rules budget unfortunately:

```
Forbidden: estimated rule cost exceeds budget by factor of 1.153434x (try simplifying the rule, or adding maxItems, maxProperties, and maxLength where arrays, maps, and strings are declared)
```